openfl/utilities/attestation/attestation_utils.py

```diff
@@ -0,0 +1,385 @@
+        """Saves the attested identity passport to a file.
+
+        Args:
+            path (str): Path to save the passport file.
+        """
+        if not os.path.exists(os.path.dirname(path)):
+            # give an error if the directory does not exist
+            raise FileNotFoundError(f"Directory {os.path.dirname(path)} does not exist")
+
+        passport = self.get_passport()
+        with open(path, "w") as f:
+            json.dump(passport, f, indent=4)
+
+    def save_token(self, path):
+        """Saves the AVS report token to a file.
+
+        Args:
+            path (str): Path to save the token file.
+        """
+        if not os.path.exists(os.path.dirname(path)):
+            # give an error if the directory does not exist
+            raise FileNotFoundError(f"Directory {os.path.dirname(path)} does not exist")
+
+        with open(path, "w") as f:
+            f.write(self.token)
+
+    def save_pubkey(self, path):
+        """Saves the public key to a file.
+
+        Args:
+            path (str): Path to save the public key file.
+        """
+        if not os.path.exists(os.path.dirname(path)):
+            # give an error if the directory does not exist
+            raise FileNotFoundError(f"Directory {os.path.dirname(path)} does not exist")
+
+        with open(path, "wb") as f:
+            f.write(self.cert)
+
+    def save_cert(self, path):
+        """Saves the certificate to a file.
+
+        Args:
+            path (str): Path to save the certificate file.
+        """
+        if not os.path.exists(os.path.dirname(path)):
+            # give an error if the directory does not exist
+            raise FileNotFoundError(f"Directory {os.path.dirname(path)} does not exist")
+
+        with open(path, "wb") as f:
+            f.write(self.cert)
+
+
+class AttestationManager:
+    """Class to manage attestation for participant enclaves.
+    This class handles the generation of SGX quotes, fetching MRENCLAVE values,
+    and obtaining attestation reports from AVS (Attestation Verification Service).
+    """
+
+    def __init__(
+        self,
+        participant_name,
+        attestation_report_path,
+        ita_api_key,
+        avs_url,
+        privkey_path=None,
+    ):
+        """Initializes the AttestationManager with the provided parameters.
+
+        Args:
+            participant_name (str): Name of the enclave.
+            attestation_report_path (str): Path to the store attestation reports.
+            ita_api_key (str): API key for ITA.
+            avs_url (str): URL for AVS.
+        """
+        self.participant_name = participant_name
+        self.attestation_report_path = attestation_report_path
+        self.ita_api_key = ita_api_key
+        self.avs_url = avs_url
+        self.privkey_path = (
+            privkey_path is not None
+            and privkey_path
+            or os.path.join(attestation_report_path, f"{participant_name}_privkey.pem")
```

Comment:
Now that I re-look at this, it is likely not safe to save the private key into the attestation report path. Let's discuss additionally how to deal with this (hard-coded path vs defaults vs other options?).